0000-route-broadcasting-protocol/0000-route-broadcasting-protocol.md

```diff
@@ -0,0 +1,160 @@
+
+Because the logs of difference could be a large number in some cases, the node may send updates separately multiple times. In that case, "from" and "to" epoch will be specified to express the range of the logs.
+
+![Epoch Logs and Its Update Request](./images/epoch.svg)
+
+The left image shows how epochs are stored. The right image shows an example of a route update request, how "from" and "to" epoch index are given, and how epochs are extracted from the logs.
+
+If the routing table ID that an epoch is based on is changed, the epoch MUST be reset to `0`, so that the counter node can track all the updates from the node.
+
+### Node Finite State Machine (FSM)
+A node is kind of an FSM that has a status of either:
+
+- IDLE
+  - Does not send route update to the counterpart node.
+- SYNC
+  - Does send route update to the counterpart node.
+
+The status can be changed by a route control request that is described below.
+
+## Protocol Detail
+### Route Control
+#### Procedure
+Route control is done in the following procedure:
+
+- A node requests IDLE or SYNC mode to the corresponded node.
+- The corresponded node changes its status if needed, and respond that the request is done.
+- If the request cannot be deserialized or interpreted as appropriate, the corresponded node responds error.
+
+#### Packet
+The request above is transferred in ILP packets in a specific manner. The manner is:
+
+- Request
+  - The `type` of the ILP packet is `ILP Prepare` (type id: 12)
+  - The `amount` of the ILP packet is `0`
+  - The `expiresAt` of the ILP packet is arbitrary
+  - The `executionCondition` of the ILP packet is `Zmh6rfhivXdsj8GLjp+OIAiXFIVu4jOzkCpZHQ1fKSU=` in Base64 format
+  - The `destination` address of the ILP packet is `peer.route.control`
+  - The `data` of the ILP packet is described at `RouteControlRequestData` in [ASN.1 definition](#asn1-definition)
+- Response
+  - The `type` of the ILP packet is `ILP Fulfill` (type id: 13)
+  - The `fulfillment` of the ILP packet is 32byte octet string all filled with zeros
+  - The `data` of the ILP packet is empty (size: 0)
+- Error
+  - The `type` of the ILP packet is `ILP Reject` (type id: 14)
+  - The `code` of the ILP packet is arbitrary, depending on the situation
+  - The `message` of the ILP packet is arbitrary, depending on the situation
+  - The `triggeredBy` of the ILP packet is the ILP address of the node where the error occuered
+  - The `data` of the ILP packet is empty (size: 0)
+
+### Route Update
+#### Procedure
+Route Update is done in the following procedure:
+
+- A node requests an update to the corresponded node with information of route update logs.
+- The corresponded node updates its routing table, and respond that the request is done.
+- If the request cannot be deserialized or interpreted as appropriate, the corresponded node responds error.
```

Comment:
It would be good to also add summary notes about what should happen between Child, Peer, and Parent nodes with regard to the overall protocol.

Reply:
Added `Link Relations and Routing` section. What do you think?